middleware/auth/auth.go

```diff
@@ -0,0 +1,45 @@
+package auth
+
+import (
+	"time"
+	"net/http"
+
+	"github.com/gorilla/mux"
+
+	"github.com/SermoDigital/jose/jws"
+	"github.com/SermoDigital/jose/crypto"
+)
+
+//Leeways for token expiration and token not-before times. Default to 5 minutes
+var expLeeway int64 = 300000000000
+var nbfLeeway int64 = 300000000000
+
+type authHandler struct {
+	handler http.Handler
+}
+
+func RequireAuthentication() mux.MiddlewareFunc {
+	return func(next http.Handler) http.Handler {
+		return AuthHandler(next)
+	}
+}
+
+func (a authHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	token, err := jws.ParseJWTFromRequest(req)
+	if err != nil {
+		//JWT Parse error - no token found, probably not authenticated
+		http.Redirect(w, req, "localhost:3000/user/login", 302)
```

Comment:
I believe that removing the "localhost:3000" and replacing it with a relative path will work as well. Like:
http.Redirect(w, req, "/user/login", 302)
Will that be fine, or should we still use the Registered URLs?

Reply:
I think registered URL's is the best practice and will make our code more flexible and maintainable moving forward.

Reply:
@dli357 @bunsenmcdubbs why are we redirecting? only the frontend should handle UI changes. The idea of having a JWT is so that the response is blocked if the token is invalid or non-existent.

Reply:
So I guess just send a 401 unauthorized instead of redirecting? That would probably be better.

Reply:
@dli357 yea 401 is much better

Reply:
Good point @Anish2. Lets do that instead. No redirects, just 401